x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/cohere/embeddings/CohereEmbeddingType.java

```diff
@@ -20,11 +24,21 @@ public enum CohereEmbeddingType {
     /**
      * Use this when you want to get back the default float embeddings. Valid for all models.
      */
-    FLOAT,
+    FLOAT(DenseVectorFieldMapper.ElementType.FLOAT),
     /**
      * Use this when you want to get back signed int8 embeddings. Valid for only v3 models.
      */
-    INT8;
+    INT8(DenseVectorFieldMapper.ElementType.BYTE),
```

Comment:
int8 will now be persisted as `byte` for the inference endpoint configuration. When we send the request to cohere we'll still use `int8`.